utils/wasm-gen/src/tests.rs

```diff
@@ -157,6 +167,245 @@ fn injecting_addresses_works() {
+            );
+            (params_config, None, Payload::default())
+        }
+        SysCallName::PayProgramRent => (
+            SysCallsParamsConfig::constant_value(0),
+            Some(MemoryWrite {
+                offset: 0,
+                content: std::iter::repeat(255).take(128).collect(),
+            }),
+            Payload::default(),
+        ),
+        _ => (
+            SysCallsParamsConfig::constant_value(0),
+            None,
+            Payload::default(),
+        ),
+    }
+}
+
+fn execute_wasm_with_syscall_injected(
+    syscall: SysCallName,
+    ignore_fallible_errors: bool,
+    params_config: SysCallsParamsConfig,
+    initial_memory_write: Option<MemoryWrite>,
+    payload: Payload,
+) -> TerminationReason {
+    use gear_backend_common::{BackendReport, Environment};
+    use gear_backend_wasmi::WasmiEnvironment;
+    use gear_core::{
+        gas::{GasAllowanceCounter, GasCounter, ValueCounter},
+        memory::AllocationsContext,
+        message::{ContextSettings, DispatchKind, IncomingDispatch, MessageContext},
+        reservation::GasReserver,
+    };
+    use gear_core_processor::{configs::PageCosts, ProcessorContext, ProcessorExternalities};
+
+    const INITIAL_PAGES: u16 = 1;
+    const INJECTED_SYSCALLS: u32 = 8;
+
+    let buf = vec![0; UNSTRUCTURED_SIZE];
+    let mut unstructured = Unstructured::new(&buf);
+
+    let mut injection_amounts = SysCallsInjectionAmounts::all_never();
+    injection_amounts.set(syscall, INJECTED_SYSCALLS, INJECTED_SYSCALLS);
+
+    let gear_config = (
+        GearWasmGeneratorConfigBuilder::new()
+            .with_memory_config(MemoryPagesConfig {
+                initial_size: INITIAL_PAGES as u32,
+                ..MemoryPagesConfig::default()
+            })
+            .with_sys_calls_config(
+                SysCallsConfigBuilder::new(injection_amounts)
+                    .with_params_config(params_config)
+                    .set_ignore_fallible_syscall_errors(ignore_fallible_errors)
+                    .build(),
+            )
+            .with_entry_points_config(EntryPointsSet::Init)
+            .with_recursions_removed(true)
+            .build(),
+        SelectableParams {
+            call_indirect_enabled: false,
+            allowed_instructions: vec![],
+            max_instructions: 0,
+            min_funcs: 1,
+            max_funcs: 1,
+        },
+    );
+
+    let module = generate_gear_program_module(&mut unstructured, gear_config)
+        .expect("failed wasm generation");
+
+    let module = gear_wasm_instrument::inject(
+        module,
+        &gear_wasm_instrument::rules::CustomConstantCostRules::new(0, 0, 0),
+        "env",
+    )
+    .unwrap();
+    let code = module.into_bytes().unwrap();
+
+    let init_msg = IncomingMessage::new(
+        MessageId::default(),
+        ProgramId::default(),
+        payload,
+        0,
+        0,
+        None,
+    );
+
+    let default_pc = ProcessorContext {
```

Comment:
what about having a separate `Default` trait impl only for tests?

Reply:
Renamed variable, `Default` impl not fits because `ProcessorContext` instantiated from params local to this `fn` and make sense only in context of this `fn`